nifi-toolkit/nifi-toolkit-cli/src/main/java/org/apache/nifi/toolkit/cli/impl/result/SimpleResultWriter.java

```diff
@@ -63,9 +61,9 @@ public void writeBuckets(List<Bucket> buckets, PrintStream output) {
 
         output.println();
 
-        final int nameLength = 30;
+        final int nameLength = buckets.stream().mapToInt(b -> b.getName().length()).max().orElse(20);
```

Comment:
I can't merge this PR at the moment since I am in the middle of some major changes, but I wanted to clarify this line... unless I am misunderstanding something, this is saying "find the max, or if nothing was then use 20". So it is not capping the length at 20, it is using 20 as a minimum which doesn't seem right.

Reply:
The `orElse()` part could be misleading, I think it can be dropped altogether, it was a legacy code.

So basically what we are trying to get is this kind of compact output (width-wise) when possible:
```
#> registry list-buckets                                                                                             
#     Name               Id                                     Description
---   ----------------   ------------------------------------   -----------
1     API First Bucket   42c47c13-5694-4b77-88c6-ef48d4b9dbc1   (empty)    
2     DCH                078cbb53-afa4-437a-ad5a-83665b3459dd   (empty)    
3     QuickImport        4d18d86e-b204-4bc0-b419-2e1818156849   (empty)    
4     Second Bucket      76dbb3e8-a812-4bb2-800a-ba3a4ec3ed72   (empty) 
```
Without it, the `Name` is too wide, even though the actual values are shorter, and the `Description` causes line wrapping and breaks the table. I.e. this code will wrap the width around the actual value, but also honor the `abbreviate` call that we have now.

Reply:
I think I'm still missing something... can you show an example output where there is a bucket with a name that is larger than 20 chars and a long description?

Reply:
Sure. Here's the output without compaction:
```
#> registry list-buckets                                                                                             

#     Name                             Id                                     Description                             
---   ------------------------------   ------------------------------------   ----------------------------------------
1     API First Bucket                 42c47c13-5694-4b77-88c6-ef48d4b9dbc1   (empty)                                 
2     DCH                              078cbb53-afa4-437a-ad5a-83665b3459dd   (empty)                                 
3     QuickImport                      4d18d86e-b204-4bc0-b419-2e1818156849   (empty)                                 
4     Second Bucket                    76dbb3e8-a812-4bb2-800a-ba3a4ec3ed72   (empty)                                 

```

And here's the one with (notice the width of the `Name` and `Desc` columns. The above output looks bad in a terminal when lines wrap:
```
#> registry list-buckets                                                                              

#     Name               Id                                     Description
---   ----------------   ------------------------------------   -----------
1     API First Bucket   42c47c13-5694-4b77-88c6-ef48d4b9dbc1   (empty)    
2     DCH                078cbb53-afa4-437a-ad5a-83665b3459dd   (empty)    
3     QuickImport        4d18d86e-b204-4bc0-b419-2e1818156849   (empty)    
4     Second Bucket      76dbb3e8-a812-4bb2-800a-ba3a4ec3ed72   (empty)   
```

Reply:
I understand the point about shrinking, but that output still doesn't show one with a long name or description. Can you run something like this first:

registry create-bucket -bn "This is a really really really really really really long name"

Then list-buckets with your changes.

Reply:
I see. It appears one does need that logic I had in the original PR. Here's the relevant piece:
```
        int nameLength = buckets.stream().mapToInt(b -> b.getName().length()).max().orElse(20);
        final int idLength = 36;
        final int descLength = 11; // 'description' word length

        nameLength = Math.min(36, nameLength);

```
The `Math.min()` was removed at some point, but it's required. The logic flow is:
- find the longest actual value
- Keep it short (we assume empty names aren't allowed) if possible
- Use the max cap (36 chars in this example) otherwise, and this is the value `abbreviate` will use.

Reply:
Ok I think I resolved this on my branch using a combination of this PR and your original PR, so this can probably be closed if you agree.